index.bs

```diff
@@ -686,32 +686,150 @@ dictionary MLOperandDescriptor {
+    1. Let |operand| be a new [=object=].
+    1. Set |operand|.{{MLOperand/[[builder]]}} to |builder|.
+    1. Set |operand|.{{MLOperand/[[kind]]}} to |kind|.
+    1. Set |operand|.{{MLOperand/[[descriptor]]}} to |desc|.
+    1. Return |operand|.
+</div>
+
+To <dfn>copy MLOperand</dfn> given |operand|, run the following steps:
+<div class=algorithm-steps>
+    1. If |operand| is not an instance of {{MLOperand}}, then throw a "{{TypeError}}" and stop.
+    1. Let |result| be a new [=object=].
+    1. Set |result|.{{MLOperand/[[builder]]}} to |operand|.{{MLOperand/[[builder]]}}.
+    1. Set |result|.{{MLOperand/[[kind]]}} to |operand|.{{MLOperand/[[kind]]}}.
+    1. Set |result|.{{MLOperand/[[descriptor]]}} to |operand|.{{MLOperand/[[descriptor]]}}.
+    1. Set |result|.{{MLOperand/[[name]]}} to |operand|.{{MLOperand/[[name]]}}.
+    1. Return |result|.
+</div>
+
+To <dfn>check dimensions</dfn> given |dimensions| and |type|, run the following steps:
+<div class=algorithm-steps>
+    1. If |dimensions| is not an array of positive numbers, return `false`;
+    1. If |dimensions|.length is 0, return `false`.
+    1. If |dimensions|.length is too large to be supported by the implementation, return `false`.
+    1. If any element of |dimensions| is not a positive number, or it is too large to be supported by the implementation given |type|, return `false`.
+    1. Return `true`.
+</div>
+
+To <dfn for="MLOperand">validate MLOperand</dfn> given |operand| and |builder|, run the following steps:
+<div class=algorithm-steps>
+    1. If |operand|.{{MLOperand/[[builder]]}} is not an instance of {{MLGraphBuilder}}, return `false`.
+    1. If |builder| is not `undefined` and is not equal to |operand|.{{MLOperand/[[builder]]}}, return `false`.
+    1. If |operand|.{{MLOperand/[[kind]]}} is not an {{MLOperandKind}}, return `false`.
+    1. Let |desc| be |operand|.{{MLOperand/[[descriptor]]}}.
+    1. If |desc| is not an [=object=] that [=implements=] {{MLOperandDescriptor}}, return `false`.
+    1. If |desc|.{{MLOperandDescriptor/dimensions}} [=map/exists=] and invoking <a>check dimensions</a> given |desc|.{{MLOperandDescriptor/dimensions}} and |desc|.{{MLOperandDescriptor/type}} returns `false`, then return `false`.
+    1. Return `true`.
+</div>
 
 ### The MLActivation interface ### {#api-mlactivation}
 
 Objects implementing the {{MLActivation}} interface represent activation function types.
 
+<div class="note">
+These activations function types are used to create other operations. One such use of this interface is for when an activation function is fused into another operation such as [[#api-mlgraphbuilder-conv2d]] or [[#api-mlgraphbuilder-batchnorm]] during a graph construction session. Such fused activation functions can provide a significant performance improvement when supported natively by the underlying implementation. This is intended as an optimization opportunity for implementers.
+</div>
```

Comment:
A reservation I have with this wording is that it doesn't define what "fuse" in this context actually means. Although the term "fusion" is commonly used, it is only used within a very narrow circle of ML framework/platform developers, and is not generally a significant concept for ML developers using those framework because it's just an implementation detail. 

However, to really explain why we need to define the `MLActivation` interface, the description should ideally focus on the role this interface plays in the construction of other operation. That can be worded without even the need to use the word "fuse" or "fusion".

Reply:
This is not in scope for this PR, it has been there earlier and I think it should be fixed separately and consistently over the whole spec.